bites_test.py

```diff
@@ -596,5 +597,31 @@ def __init__(self, name, score):
     self.assertEqual(print_mock.call_args_list[1], mock.call("luigi: 9\n"))
     print_patcher.stop()
 
+class PlayFullGameTest(unittest.TestCase):
+  @patch('bites.Bites.take_all_turns')
+  def test_play_full_game_calls_take_all_turns(self, mock):
+  # test 82
+    bites_game = Bites([], {}, [])
+    bites_game.play_full_game()
+    self.assertTrue(mock.called)
+
+  @patch('bites.Bites.print_scores')
+  def test_play_full_game_calls_print_scores(self, mock):
+  # test 83
+    class FakePlayer():
+      def __init__(self, name, score):
+        self.name = name
+        self.score = score
+      
+      def take_turn(self, trail, ant_positions, anthill):
+        return (trail, ant_positions, anthill)
+
+    fake_mario = FakePlayer("mario", 3)
+    fake_luigi = FakePlayer("luigi", 9)
+    players = [fake_mario, fake_luigi]
+    bites_game = Bites([], {}, players)
+    bites_game.play_full_game()
+    self.assertTrue(mock.called)
+
```

Comment:
To take this testing a step further, you might want to test that the two functions (take_all_turns, print_scores) are called in the correct order. This is important because we want the scores to be printed at the end of the game, not at the start. This would involve creating two mocks with two decorators, and then checking that one was called before the other: https://stackoverflow.com/questions/32463321/how-to-assert-method-call-order-with-python-mock